core/src/main/java/org/testcontainers/containers/output/BaseConsumer.java

```diff
@@ -0,0 +1,17 @@
+package org.testcontainers.containers.output;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.function.Consumer;
+
+public abstract class BaseConsumer<S extends BaseConsumer<S>> implements Consumer<OutputFrame> {
```

Comment:
Just a question, why `S`?

Reply:
Initially I named this like `SELF`, but Codacity told me, that type parameters should be length one character. And I have trim this name to `S` :smile:

Reply:
If this has the same semantics than the `SELF` used in `GenericContainer`, I'd prefer `SELF` (and ignore the warning :slightly_smiling_face:).